pkg/function/scale_workload.go

```diff
@@ -80,15 +82,33 @@ func (s *scaleWorkloadFunc) Exec(ctx context.Context, tp param.TemplateParams, a
 	}
 	switch strings.ToLower(kind) {
 	case param.StatefulSetKind:
-		return nil, kube.ScaleStatefulSet(ctx, cli, namespace, name, replicas, waitForReady)
+		count, err := kube.StatefulSetReplicas(ctx, cli, namespace, name)
+		if err != nil {
+			return nil, err
+		}
+		return map[string]interface{}{
+			outputArtifactOriginalReplicaCount: count,
+		}, kube.ScaleStatefulSet(ctx, cli, namespace, name, replicas, waitForReady)
```

Comment:
I'd suggest the following change, since it's a best practice to return an empty value when an error occurs. It clarifies the intent and also it maintains consistency with the standard library.

Also I think the result makes no sense in case of scaling failure, am I right ?

```suggestion
		count, err := kube.StatefulSetReplicas(ctx, cli, namespace, name)
		if err != nil {
			return nil, err
		}
		
		err = kube.ScaleStatefulSet(ctx, cli, namespace, name, replicas, waitForReady)
		if err != nil {
			return nil, err
		}
		
		return map[string]interface{}{
			outputArtifactOriginalReplicaCount: count,
		}, nil
```

The same pays for following changes too.

Reply:
I can do it, but in kanister generally we have been just returning the method if the return type matches. Here are some examples

https://github.com/kanisterio/kanister/blob/master/pkg/phase.go#L51
https://github.com/kanisterio/kanister/blob/master/pkg/phase.go#L97
https://github.com/kanisterio/kanister/blob/master/pkg/app/cockroachdb.go#L298
https://github.com/kanisterio/kanister/blob/master/pkg/blockstorage/vmware/vmware.go#L338